Tests/scripts/test_modeling_rule_report.py

```diff
@@ -98,28 +97,10 @@ def generate_description_for_test_modeling_rule(ci_pipeline_id: str,
     return description
 
 
-@tenacity.retry(
-    wait=tenacity.wait_exponential(multiplier=1, min=2, max=10),
-    stop=tenacity.stop_after_attempt(2),
-    reraise=True,
-    retry=tenacity.retry_if_exception_type(JIRAError),
-    before_sleep=tenacity.before_sleep_log(logging.getLogger(), logging.DEBUG)
-)
-def search_in_jira_ticket_for_test_modeling_rule(jira_server: JIRA, summary: str) -> Issue | None:
-    ticket_summary = generate_ticket_summary(summary)
-    jql_query = generate_query(ticket_summary)
-    search_issues: ResultList[Issue] = jira_server.search_issues(jql_query)  # type: ignore[assignment]
-    if search_issues:
-        summary_lower = summary.lower()
-        for issue in search_issues:
-            if summary_lower in issue.get_field("summary").lower():
-                return issue
-    return None
-
-
 def calculate_test_modeling_rule_results(test_modeling_rules_results_files: dict[str, Path],
-                                         jira_server: JIRA | None = None
+                                         issues: dict[str, list[Issue]] | None = None
                                          ) -> tuple[dict[str, dict[str, TestSuite]], dict[str, Issue], set[str]]:
+    issues = issues or {}
```

Comment:
Why not use a default value in the definition of the param in the method definition?

Reply:
https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/ 
mutable arguments are dangerours in python.